src/test/java/bio/terra/service/iam/sam/SamIamTest.java

```diff
@@ -57,4 +133,274 @@ public void testExtractErrorMessageDeepNested() {
 
         assertThat(SamIam.extractErrorMessage(errorReport)).isEqualTo("FOO: BAR: (BAZ1, BAZ2: QUX)");
     }
+
+    @Test
+    public void testIgnoresNonUUIDResourceName() throws ApiException, InterruptedException {
+        final String goodId = UUID.randomUUID().toString();
+        final String badId = "badUUID";
+        when(samResourceApi.listResourcesAndPolicies(IamResourceType.SPEND_PROFILE.getSamResourceName()))
+            .thenReturn(List.of(
+                new ResourceAndAccessPolicy().resourceId(goodId),
+                new ResourceAndAccessPolicy().resourceId(badId)
+            ));
+
+        List<UUID> uuids = samIam.listAuthorizedResources(userReq, IamResourceType.SPEND_PROFILE);
+        assertThat(uuids).containsExactly(UUID.fromString(goodId));
+    }
+
+    @Test
+    public void testAuthorization() throws ApiException, InterruptedException {
+        when(samResourceApi.resourcePermissionV2(
+            IamResourceType.SPEND_PROFILE.getSamResourceName(),
+            "my-id",
+            IamAction.READ_DATA.toString()
+        )).thenReturn(true);
+        when(samResourceApi.resourcePermissionV2(
+            IamResourceType.SPEND_PROFILE.getSamResourceName(),
+            "my-id",
+            IamAction.ALTER_POLICIES.toString()
+        )).thenReturn(false);
+        assertThat(samIam.isAuthorized(userReq, IamResourceType.SPEND_PROFILE, "my-id", IamAction.READ_DATA))
+            .isTrue();
+        assertThat(samIam.isAuthorized(userReq, IamResourceType.SPEND_PROFILE, "my-id", IamAction.ALTER_POLICIES))
+            .isFalse();
+    }
+
+    @Test
+    public void testGetStatus() throws ApiException {
+        when(samStatusApi.getSystemStatus()).thenReturn(new SystemStatus()
+            .ok(true)
+            .systems(Map.of("GooglePubSub", Map.of("ok", true)))
+        );
+        assertThat(samIam.samStatus()).isEqualTo(
+            new RepositoryStatusModelSystems()
+                .ok(true)
+                .message("{GooglePubSub={ok=true}}")
+        );
+    }
+
+    @Test
+    public void testGetStatusException() throws ApiException {
+        when(samStatusApi.getSystemStatus()).thenThrow(new ApiException("BOOM!"));
+        assertThat(samIam.samStatus()).isEqualTo(
+            new RepositoryStatusModelSystems()
+                .ok(false)
+                .message("Sam status check failed: bio.terra.service.iam.exception.IamInternalServerErrorException: " +
+                    "BOOM!")
+        );
+    }
+
+    @Test
+    public void testGetUserInfo() throws ApiException {
+        final String userSubjectId = "userid";
+        final String userEmail = "a@a.com";
+        when(samUsersApi.getUserStatusInfo())
+            .thenReturn(new org.broadinstitute.dsde.workbench.client.sam.model.UserStatusInfo()
```

Comment:
How does this differ from `bio.terra.model.UserStatusInfo`?

Reply:
It's actually very similar but we didn't want to return objects that we don't control